www/docs/server/context.md

```diff
@@ -60,11 +62,88 @@ export const router = t.router;
 
 /**
  * Unprotected procedure
- **/
+ */
 export const publicProcedure = t.procedure;
 
 /**
  * Protected procedure
- **/
+ */
 export const protectedProcedure = t.procedure.use(isAuthed);
-```
\ No newline at end of file
+```
+
+## Inner and outer context
+
+In some scenarios it could make sense to split up your context into "inner" and "outer" functions.
+
+**Inner context** is where you define context which doesn’t depend on the request, e.g. your database connection. You can use this function for integration testing or [SSG helpers](ssg-helpers), where you don’t have a request object. Whatever is defined here will **always** be available in your procedures.
+
+**Outer context** is where you define context which depends on the request, e.g. for the user's session. Whatever is defined here is not available in all procedures, but only for procedures that are called via HTTP.
+
+### Example for inner & outer context
+
+```ts
+import type { inferAsyncReturnType } from '@trpc/server';
+import type { CreateNextContextOptions } from '@trpc/server/adapters/next';
+
+/** 
+ * Defines your inner context shape.
+ * Add fields here that the inner context brings.
+ */
+interface CreateInnerContextOptions extends Partial<CreateNextContextOptions> {
```

Comment:
I don't think this is correct. Prisma definitely doesn't belong in here - I guess you _could_ pass it from outer context, but this example isn't, and I don't see a reason to. It's usually just imported at the top of the file and then used inside `createContextInner`.

Also not sure about opts (ie req/res), they type is technically not incorrect as it's a partial, but why type it like this when req and res aren't actually being passed to `createContextInner` from `createContext`?

Reply:
Makes sense, used a "session" instead.